probe-rs/src/probe/stlink/mod.rs

```diff
@@ -1482,30 +1358,95 @@ where
 {
     type Error = DebugProbeError;
 
-    fn read_ap_register(&mut self, port: impl Into<AP>, register: R) -> Result<R, Self::Error> {
-        self.read_ap_register(port, register)
+    /// Read the given register `R` of the given `AP`, where the read register value is wrapped in
+    /// the given `register` parameter.
+    fn read_ap_register(&mut self, port: impl Into<AP>, _register: R) -> Result<R, Self::Error> {
+        log::debug!("Reading register {}", R::NAME);
+        self.select_ap_and_ap_bank(port.into().port_number(), R::APBANKSEL)?;
+
+        let result = self.probe.read_register(
+            PortType::AccessPort(u16::from(self.state.current_apsel)),
+            u16::from(R::ADDRESS),
+        )?;
+
+        log::debug!("Read register    {}, value=0x{:08x}", R::NAME, result);
```

Comment:
Put this on trace? :)

Reply:
This is unchanged from before, and also done like this in the other ARM probes.

Reply:
Yeah, I just saw it and as we discussed once that we should refine log levels and adam already did some refinements, I thought it would be a good thing to do, but I guess we just go ahead for now :)